extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -2196,10 +2198,19 @@ private boolean isNotTemplateLocatorImpl(AnnotationTarget locateTarget) {
         return !targetImplTemplateLocator;
     }
 
-    private static boolean isImplementorOf(ClassInfo target, DotName interfaceName) {
-        for (DotName name : target.asClass().interfaceNames()) {
-            if (name.equals(interfaceName)) {
-                return true;
+    private static boolean isImplementorOf(ClassInfo target, DotName interfaceName, IndexView index) {
+        ClassInfo current = target;
+        while (current != null) {
+            for (DotName name : current.interfaceNames()) {
+                if (name.equals(interfaceName)) {
+                    return true;
+                }
+            }
+            DotName superName = current.superName();
+            if (superName != null && !superName.equals(DotName.OBJECT_NAME)) {
+                current = index.getClassByName(superName);
+            } else {
+                current = null;
```

Comment:
Well, this fixes my problem, but you're still forgetting the super-interfaces or super-interfaces. Don't we already have existing sub-typing utilities for Jandex in Quarkus? We must have, no?

Reply:
You're right. I did not think of this use case :shrug:.

> Don't we already have existing sub-typing utilities for Jandex in Quarkus? We must have, no?

I did not find anything in the io.quarkus.deployment.util.JandexUtil`...

Reply:
Should be fixed now. The logic got extracted into the `io.quarkus.qute.deployment.Types#isImplementorOf()` method so that we could move it up if needed...